src/harness/testcases/WINNF_FT_S_WDB_testcase.py

```diff
@@ -205,24 +246,48 @@ def test_WINNF_FT_S_WDB_2(self, config_filename):
     # Send registration requests.
     registration_responses = self._sas.Registration(registration_requests)['registrationResponse']
 
-    # Check registeration response,
+    # Check registration response,
     # responseCode should be 103 (INVALID_VALUE).
     for registration_response in registration_responses:
       self.assertEqual(registration_response['response']['responseCode'], 103)
 
     del registration_responses
 
-    # TODO
     # Step 2: Create a CPI database which includes CPI user with credentials
     # matching those used to create the request in Step 1.
+    cpi_database = DatabaseServer('CPI Database',config['cpiDatabaseConfig']['hostName'],
+                                  config['cpiDatabaseConfig']['port'])
+
+    # Start CPI database server.
+    cpi_database.start()
+
+    # Generate the CSV file and write the content of it.
+    config_keys = config['cpiDatabaseConfig']['cpis'][0].keys()
+
+    with open(config['cpiDatabaseConfig']['cpis'][0]['publicKeyIdentifierFile'], 'w+') as output_file:
+        dict_writer = csv.DictWriter(output_file, config_keys)
```

Comment:
This will not write them in a deterministic order, since line 265 is not deterministic. (Moreover, the dict keys are not the same as the column names.) Please hard-code the column names.

Reply:
Fixed as per suggestion